### ISSUE TITLE
Stream output to any source with shell script

### ISSUE THREAD

**Issue by sthgrau** [sparkfun/phant#237]
I want to be able to send records to a mysql database, but I was not able to find a stream output module for mysql. I decided I could use the stderr from the validator, parsing it and inserting it into the database. Since it is an interpreted script, you can change it to output whatever you need without needing to learn the project structure or api. Note, the following implementation is dependent on #236 to remove duplicates and change the output to parseable json.

The following python script reads in from stdin, when it seems like has a complete json record, it parses it, then grabs the field names and values and creates an sql statement. This output is then piped to mysql to insert into the database. NOTE: change the tableName variable value to the tablename in mysql and if you are running without the shebang (ie 'python somedudesscript.py'), note the '-u' flag for unbuffered stdout (ie 'python -u somedudesscript.py')

To run, stderr from phant needs to be folded into stdout, then piped to the python script, then piped to mysql.
`sudo ./.bin/serve 2>&1 | ~/bin/somedudesscript.py | mysql -u USER --password=PASS YOURDATABASE`

```
#!/usr/bin/python -u

import sys
import re
import json
import string
tableName="YOURTABLE"
lines=""
while True:
  line=sys.stdin.readline().strip()
  if line == "":
    sys.exit(0)
  if re.search('^{', line):
   lines=""
  lines+=line
  if re.search('}$', line):
   try:
     stuff=json.loads(lines)
   except:
     sys.stderr.write("oops\n")
   lines=""
   myvals=[]
   for myk in stuff.keys():
      myvals.append(stuff[myk])

   print "insert into %s (%s) values (%s);" % (tableName, string.join(stuff.keys(),','),'"' + string.join(myvals,'","') + '"')

```

**Comment by bboyho** [comment]
### Phant is No Longer in Operation

Unfortunately Phant, our data-streaming service, is no longer in service and will be discontinued. The system has reached capacity and, like a less-adventurous Cassini, has plunged conclusively into a fiery and permanent retirement. There are several other maker-friendly, data-streaming services and/or IoT platforms available as alternatives. The three we recommend are Blynk, ThingSpeak, and Cayenne. You can read our blog post [ [https://www.sparkfun.com/news/2413](https://www.sparkfun.com/news/2413) ] on the topic for an overview and helpful links for each platform.
